### ISSUE TITLE
The Got stream does not emit "error" or "response" events if a redirect response was destroyed

### ISSUE THREAD

**Issue by mikeconley12** [sindresorhus/got#1326]
#### Describe the bug

 - Node.js version: v12.16.3
 - OS & version: Ubuntu 18.04.4

The Got stream does not emit "error" or "response" events if a redirect response was destroyed.
See [PoC](https://repl.it/repls/DizzyJudiciousSet#index.js)

#### Actual behavior

The PoC prints:

```
redirectResponse.destroy()
```

#### Expected behavior

The PoC should print:

```
redirectResponse.destroy()
write EPROTO 140306412820288:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:
```

#### Code to reproduce

```js
const Koa = require("koa");
const stream = require("stream");
const util = require("util");

const finished = util.promisify(stream.finished);
const app = new Koa();
const got = require("got").extend({
  timeout: 20000
});

app.use(async ctx => {
  ctx.set("Location", "https://localhost:3000");
  ctx.set("Content-Encoding", "gzip");
  ctx.status = 301;
});

app.listen(3000);

(async () => {
  const requestStream = got
    .stream("http://localhost:3000")
    .on("redirect", redirectResponse => {
      console.log("redirectResponse.destroy()");
      redirectResponse.destroy();
    })
    .on("response", response => console.log("response.requestUrl", response.requestUrl));

  try {
    await finished(requestStream);
    console.log("DONE");
  } catch (error) {
    console.error(error.message);
  }
})();
```

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment]
Fixed in `got@11.4.0`.

**Comment by mikeconley12** [comment]
Thank you!
